### DISCUSSION TITLE
Error validating datasource `db`: the URL must start with the protocol `prisma://`

### DISCUSSION THREAD

**Discussion by MarioGith** [prisma/prisma#23533] [Q&A]
### Question

Hi everyone,

I'm truly struggling this morning using the Prisma library. When I'm in my local dev env, I've no issue using prisma and querying my PostgreSQL database.

But when I use Vercel (local or production), I've this message :

```
  Invalid `prisma.user.findUnique()` invocation:
    Error validating datasource 'db': the URL must start with the protocol 'prisma://'
    [...]
    code: 'P6001',
    clientVersion: '5.9.1',
    meta: { modelName: 'User' }
 ```

Here is my _schema.prisma_:

```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["fullTextIndex", "fullTextSearch"]
}

datasource db {
  provider  = "postgresql"
  url       = env("POSTGRES_PRISMA_URL")
  directUrl = env("POSTGRES_URL_NON_POOLING")
}
```

And here are my _.env_ on Vercel and local:

```
POSTGRES_PRISMA_URL="postgres://[...]"
POSTGRES_URL_NON_POOLING="postgres://[...]"
```

And here is my _db.ts_:

```
import { PrismaClient } from '@prisma/client'
export const db = globalThis.prisma || new PrismaClient()
if (process.env.NODE_ENV !== 'production') globalThis.prisma = db
```

https://github.com/prisma/prisma/discussions/21355 already discussed about the edge import issue, but this is not case in this issue.

I clearly don't understand why the issue is happening on Vercel production and even on local when:
`npx vercel build`
`npx vercel dev`

I've already tried to rollback from v5.11 to v5.9, but nothing changed.

Do you have any ideas or clues ?

Thank you

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["fullTextIndex", "fullTextSearch"]
}

datasource db {
  provider  = "postgresql"
  url       = env("POSTGRES_PRISMA_URL")
  directUrl = env("POSTGRES_URL_NON_POOLING")
}
```

```ts
import { PrismaClient } from '@prisma/client'
export const db = globalThis.prisma || new PrismaClient()
if (process.env.NODE_ENV !== 'production') globalThis.prisma = db
```

- OS: macOS - M3
- Database: Vercel PostgreSQL
- Node.js version: v21.6.2
- Primsa version: 5.11
- Prisma client version: 5.9.1 | 5.11

**Comment by ludralph** [comment]
Hi @MarioGith 👋 

Thank you for raising this question.

Are you trying to use [Accelerate](https://www.prisma.io/data-platform/accelerate)? Also, did you follow the steps mentioned [here](https://www.prisma.io/docs/orm/prisma-client/deployment/edge/deploy-to-vercel#vercel-postgres) to deploy your application to vercel?

**Reply by MarioGith** [reply]
Hi @ludralph,

Thank you for your answer ! At first, Accelerate was used, but I'm trying to remove it for the moment. 

For the issue, I think I'm missing the neon adapter in my setup. I'll test this and let you know, and if this solve it, sorry for misreading I've done on the doc.

**Reply by warmachine028** [reply]
thanks @ludralph

**Comment by oknoorap** [comment]
@ludralph I'm using normal database url, and not using accelerate, why it's throwing an error: the URL must start with the protocol `prisma://` or `prisma+postgres://`, I'm struggling right now.

my stack is turborepo + nestjs

**Reply by resatDev** [reply]
@oknoorap if you are generating the prisma with --no-engine parameter, the client is created by the Prisma Accelerate rules. So, necessary engines for your working & production environments will not be configured as well (Like Prisma MacOS, Prisma Linux etc.).

Why?
With Accelerate, the bundle size is decreasing, queries are caching with their data etc.

I could solve my problem with getting rid of --no-engine flag in my application;

For Example;

❌ Wrong:
`yarn prisma generate --schema=prisma/db1/schema.prisma` --no-engine

✅ Correct:
`yarn prisma generate --schema=prisma/db1/schema.prisma`

Before implementing this, make sure that the .next folder was deleted.

**Reply by Abdssamie** [reply]
this fixed it for me. I previously thought it was a caching issue but it was that all along
